internal/resolution/variable_sources/olm/olm_test.go

```diff
@@ -109,15 +130,17 @@ var _ = Describe("OLMVariableSource", func() {
 				out = append(out, variable.BundleEntity().Entity)
 			}
 			return out
-		}, Equal([]*input.Entity{
+		}, ConsistOf([]*input.Entity{
```

Comment:
> This sounds like an important question to resolve at the public-facing level - if the order is not guaranteed, it should be noted in the API docs.

We are not defining an API here - we are implementing a contract/interface required by Deppy. Solver will be calling `GetVariables` and as far as I see it doesn't require variable s to be sorted in a specific way.

https://github.com/operator-framework/deppy/blob/9a7655cd364b3900d394f81a752731200e2d01a4/pkg/deppy/input/variable_source.go#L9-L12

IMO nether sorting variables nor documenting ordering on the operator-controller side is necessary.

Reply:
> Solver will be calling `GetVariables` and as far as I see it doesn't require variable s to be sorted in a specific way.

Want to clarify: there are some places where the order is very important for the resolution process. For example, we want to select latest possible version for the solution.

But I think here it is not important. However thinking about debuggability - it still might be a good idea to have some stable order.

Reply:
I think I was partially wrong [here](https://github.com/operator-framework/operator-controller/pull/245/files#r1213446235):

> `olmVariableSource.GetVariables` now returns in different order. This is mostly due to a fake client: `sigs.k8s.io/controller-runtime/pkg/client/fake` doesn't guarantee stable order.

I looked into this a bit more. It looks like the fake client does not guarantee order (at least I do not see anything about this in docs), but de facto it does sorting (we hit [this code](https://github.com/kubernetes/client-go/blob/5a019202120ab4dd7dfb3788e5cb87269f343ebe/testing/fixture.go#L493-L501) eventually).

So in the test we add `prometheus` and then `packageA` into the fake, but when `GetVariables` calls the client - it returns `packageA` first and `prometheus` because the result list gets sorted.

Our options here:
* Change ordering of the expected results taking undocumented sorting of fake client into account. This will make the diff minimal. But I think this is not ideal - it will lead to a hard to understand failure, if underlying implementation of the fake client changes.
* Do sorting in `GetVariables` after calling the client. This is only required for making the test simplier. I think it is not right to introduce sorting in the code which we test just to make the test code a little bit simplier.
* Change tests (what we currently have in this PR) to make sure that we do not rely on order provided by the fake client. I think this is the best approach given that we do not need to guarantee order from `GetVariables`.
* Create a fake client which maintains order of objects provided in the constructor when returning a `List` result. Not sure this has any benefit: we will have to introduce extra testing code to avoid changing existing testing code (assertions).

> However thinking about debuggability - it still might be a good idea to have some stable order.

Since we now know that fake client returns in deterministic order (not documented, but still) - I think there is little to no benefit in sorting.

Reply:
If sort order is immaterial except for testing, I agree that your third option is the best.